perception/autoware_simple_object_merger/src/simple_detected_object_merger_node.cpp

```diff
@@ -38,18 +38,31 @@ void SimpleDetectedObjectMergerNode::approximateMerger(
   const DetectedObjects::ConstSharedPtr & object_msg0,
   const DetectedObjects::ConstSharedPtr & object_msg1)
 {
-  transform_ = transform_listener_->get_transform(
-    node_param_.new_frame_id, object_msg0->header.frame_id, object_msg0->header.stamp,
-    rclcpp::Duration::from_seconds(0.01));
-  DetectedObjects::SharedPtr transformed_objects0 =
-    getTransformedObjects(object_msg0, node_param_.new_frame_id, transform_);
-
-  // input1
-  transform_ = transform_listener_->get_transform(
-    node_param_.new_frame_id, object_msg1->header.frame_id, object_msg1->header.stamp,
-    rclcpp::Duration::from_seconds(0.01));
-  DetectedObjects::SharedPtr transformed_objects1 =
-    getTransformedObjects(object_msg1, node_param_.new_frame_id, transform_);
+  DetectedObjects::SharedPtr transformed_objects0;
+  if (node_param_.new_frame_id == object_msg0->header.frame_id) {
```

Comment:
memo: tf2 core has early return but it checks tf tree, so this statement is valuable.